scanner/engine/master.cpp

```diff
@@ -521,10 +522,25 @@ class MasterImpl final : public proto::Master::Service {
     return grpc::Status::OK;
   }
 
+  grpc::Status Shutdown(grpc::ServerContext *context,
+                        const proto::Empty *empty,
+                        Result *result) {
+    trigger_shutdown_ = true;
+    for (auto& w : workers_) {
+      grpc::ClientContext ctx;
+      proto::Empty empty;
+      proto::Result result;
+      w->Shutdown(&ctx, empty, &result);
+    }
+    result->set_success(true);
```

Comment:
Why would we log fatal if we are shutting down though? Maybe the worker failed (due to a SEGFAULT or something) and so the connection is no longer valid.